tools/pepsi/src/completions.rs

```diff
@@ -0,0 +1,98 @@
+use clap::{Args, Command};
+use clap_complete::{generate, Shell};
+use color_eyre::Result;
+use regex::Regex;
+
+use crate::aliveness::completions as complete_naos;
+
+#[derive(Args)]
+pub struct Arguments {
+    #[arg(long, hide = true)]
+    complete_naos: bool,
+    #[clap(name = "shell")]
+    pub shell: clap_complete::shells::Shell,
+}
+
+pub async fn completions(arguments: Arguments, mut command: Command) -> Result<()> {
+    if arguments.complete_naos {
+        let naos = complete_naos().await?;
+
+        let separator = match arguments.shell {
+            Shell::Bash => ' ',
+            Shell::Fish => '\n',
+            Shell::Zsh => '\n',
+            _ => ' ',
+        };
+
+        for nao in naos {
+            print!("{nao}{separator}");
+        }
+        return Ok(());
+    }
+
+    let mut static_completion = Vec::new();
+    generate(
+        arguments.shell,
+        &mut command,
+        "pepsi",
+        &mut static_completion,
+    );
+
+    let static_completions = String::from_utf8(static_completion)?;
+    dynamic_completions(arguments.shell, static_completions);
+    Ok(())
+}
+
+fn dynamic_completions(shell: Shell, static_completions: String) {
+    let completion_cmd = format!("pepsi completions --complete-naos {shell}");
+
+    match shell {
+        Shell::Bash => {
+            let re = Regex::new("(?:<NAOS>|\\[NAOS\\])...").unwrap();
+            let completions = re.replace_all(&static_completions, format!("$({completion_cmd})"));
+            print!("{completions}")
+        }
+        Shell::Fish => {
+            const COMPLETION_SUBCOMMANDS: [(&str, &str); 11] = [
+                ("aliveness", ""),
+                ("hulk", ""),
+                ("logs", "delete"),
+                ("logs", "downloads"),
+                ("postgame", ""),
+                ("poweroff", ""),
+                ("reboot", ""),
+                ("upload", ""),
+                ("wireless", "list"),
+                ("wireless", "set"),
+                ("wireless", "status"),
```

Comment:
👀 somehow github dropped my comment here when submitting the review...

This list is missing `shell` and `pregame`.
Not quite sure about pregame though as it requires `robot:player_number` pairs.
It might still be a good idea to at least complete the player number though, what do you think?

Reply:
You are right, I missed `shell` since its just a single NAO address. For `pregame` it might be a good idea to suggest e.g. `27:`, i.e. with colon. Since the amount of player numbers is limited, we could also suggest all combinations, i.e. `27:1`, `27:2`, `27:3`, `27:4` and `27:5` for the above example.

Both would add a bit of complexity, if this is not desired, we could just suggest the number. Which one do you prefer?

Reply:
I think the version with the trailing colon would be ideal. That increases ease of use while still indicating that the argument is incomplete.
Suggesting every combination can be quite noisy considering up to 12 robots and 7 player numbers.

Reply:
Alright, done.